### PR TITLE
Allow using custom client

### PR DESCRIPTION
**Pull request by RossyWhite**
It would be useful to make it possible to use custom client. (to customize networking options)

### CODE REVIEW

**Comment by RossyWhite** [comment]
@gsquire could you please review it?

**Comment by gsquire** [comment]
Hey, I'm currently on vacation so I won't get to this until next week. Sorry!

File: src/v3.rs [hunk 1]
```diff
@@ -217,6 +217,17 @@ impl Sender {
         self.host = host.into();
     }
 
+    /// Sets client to use for the API. This is useful if you want to customize the client.
+    pub fn set_client(&mut self, client: Client) {
```
Comment (gsquire) [inline src/v3.rs:221]: Maybe it would be better to allow `new` to accept an `Option<Client>` and just unwrap it or use a default one.

Comment (RossyWhite) [reply src/v3.rs]: @gsquire 

Is this what you meant? https://github.com/gsquire/sendgrid-rs/pull/114/commits/b5ea71350dc8bea470c36e765fee263ddff0aa83

Comment (gsquire) [reply src/v3.rs]: Exactly. We'll need a `new_blocking` function as well so users can configure that client. Otherwise this LGTM.

Comment (RossyWhite) [reply src/v3.rs]: Added! @gsquire 

https://github.com/gsquire/sendgrid-rs/pull/114/commits/62358ce9320534db0cb8c98ceefcb73f4d833275

**Comment by gsquire** [comment]
Thanks! I will publish this as version 0.22.0 shortly.

**Pull request merged**